Objects/moduleobject.c

```diff
@@ -819,68 +872,109 @@ _Py_module_getattro_impl(PyModuleObject *m, PyObject *name, int suppress)
         Py_DECREF(getattr);
         return result;
     }
+
+    if (suppress == 1) {
+        return NULL;
+    }
     if (PyDict_GetItemRef(m->md_dict, &_Py_ID(__name__), &mod_name) < 0) {
         return NULL;
     }
```

Comment:
It would be nice to have a comment that makes it clear that we're handling the attr-not-found case.

```suggestion

    // The attribute was not found.  We make a best effort at a useful error message,
    // but only if we're no suppressing AttributeError.
    
    if (suppress == 1) {
        return NULL;
    }
    if (PyDict_GetItemRef(m->md_dict, &_Py_ID(__name__), &mod_name) < 0) {
        return NULL;
    }
```